src/main/java/org/folio/copycat/RecordImporter.java

```diff
@@ -250,19 +252,44 @@ void setStoragePollIterations(int cnt) {
     });
   }
 
+  Boolean getPollingSucceeded() {
+    return pollingSucceeded;
+  }
+
+  Future<List<String>> end(List<String> instances, long updatePoll) {
+    return post(null, true)
+      .compose(
+        x -> {
+          if (!instances.isEmpty()) {
+            Promise<List<String>> promise = Promise.promise();
+            vertx.setTimer(updatePoll, w -> promise.complete(instances));
+            return promise.future();
+          }
+          return getSourceRecords(1)
+            .compose(
+              ok -> {
+                pollingSucceeded = true;
+                return Future.succeededFuture(ok);
+              },
+              cause -> {
+                pollingSucceeded = false;
+                log.warn("Polling failed and ignored: {}", cause.getMessage(), cause);
+                return Future.succeededFuture(Collections.emptyList());
+              });
+        }
+      ).onComplete(x -> client.close());
+  }
+
   /**
    * end importing.
-   * @return async result with list of instances.
+   * @param instances known instances (empty for import, non-empty for overlay)
+   * @return list updated/imported instances.
    */
-  public Future<List<String>> end() {
-    return post(null, true)
-      .compose(x ->
-        getSourceRecords(1)
-          .recover(cause -> {
-            log.warn("Polling failed and ignored: {}", cause.getMessage(), cause);
-            return Future.succeededFuture(Collections.emptyList());
-          })
-      )
-      .onComplete(x -> client.close());
+  public Future<List<String>> end(List<String> instances) {
+    return end(instances, UPDATE_POLL);
```

Comment:
Does this hardwiring mean that it will _always_ take five seconds to return from an update, even if it happens immediately?

Reply:
Yes. It's an event queue system as you know .. We don't have a clue when the instance record is propagated to instances